Parser/pegen/pegen.c

```diff
@@ -25,6 +25,21 @@ init_normalization(Parser *p)
     return 1;
 }
 
+int
+_PyPegen_check_barry_as_flufl(Parser *p){
```

Comment:
PEP 7
```suggestion
_PyPegen_check_barry_as_flufl(Parser *p) {
```

Also this function could use a comment explaining what it returns. Apparently nonzero is an error, meaning either `<>` detected in Guido mode or `!=` detected in Barry mode, so zero means "okay", except it also could mean "exception". (I guess some later check will eventually detect the exception.)

Reply:
> (I guess some later check will eventually detect the exception.)

In the exception path, we can return anything IIRC because upstream code will catch that and propagate it. The return value is only used to signal if the subrule for the token should return NULL or the token (indicating that it parses).

Reply:
Note that 0 is *not* an error here, it's success. (All this is confused by using `strcmp()`, which returns 0 for equality.) What happens if you return -1 for errors?

Reply:
> Maybe is best to always return an error (0) even after the raise.

Sorry, this should have been: "return an error (nonzero) even after the raise"

> What happens if you return -1 for errors?

The parser reports an error because the sub-rule will return `NULL` for the token